src/System.Text.Json/tests/Serialization/Stream.WriteTests.cs

```diff
@@ -149,6 +150,112 @@ public static async Task WritePrimitivesAsync()
             int i = await JsonSerializer.DeserializeAsync<int>(stream, options);
             Assert.Equal(1, i);
         }
+
+        private class Session
+        {
+            public int Id { get; set; }
+            public string Title { get; set; }
+            public virtual string Abstract { get; set; }
+            public virtual DateTimeOffset? StartTime { get; set; }
+            public virtual DateTimeOffset? EndTime { get; set; }
+            public TimeSpan Duration => EndTime?.Subtract(StartTime ?? EndTime ?? DateTimeOffset.MinValue) ?? TimeSpan.Zero;
+            public int? TrackId { get; set; }
+        }
+
+        private class SessionResponse : Session
+        {
+            public Track Track { get; set; }
+            public List<Speaker> Speakers { get; set; } = new List<Speaker>();
+        }
+
+        private class Track
+        {
+            public int Id { get; set; }
+            public string Name { get; set; }
+        }
+
+        private class Speaker
+        {
+            public int Id { get; set; }
+            public string Name { get; set; }
+            public string Bio { get; set; }
+            public virtual string WebSite { get; set; }
+        }
+
+        [Theory]
+        [InlineData(0)]
+        [InlineData(10)]
+        [InlineData(1000)]
+        [InlineData(4000)]
+        [InlineData(8000)]
+        [InlineData(16000)]
+        public static async Task LargeJsonFile(int bufferSize)
+        {
+            // Build up a large list to serialize.
+            var list = new List<SessionResponse>();
+            for (int i = 0; i < 100; i++)
```

Comment:
nit: factor 100 into constant since it is reused for test assertions.

Reply:
Sounds good. Will address in my next PR to avoid CI restart.